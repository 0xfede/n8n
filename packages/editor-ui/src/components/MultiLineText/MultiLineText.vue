<script lang="ts" setup>
/**
 * Component that splits the given text by new lines and renders
 * each line in a separate span element joined by a line break.
 */
import { computed } from 'vue';

const props = defineProps<{
	text: string;
}>();

const lines = computed(() => {
	return props.text ? props.text.split('\n') : [];
});
</script>

<template>
	<div>
		<span v-for="(line, index) in lines" :key="index">
			{{ line }}<br v-if="index < lines.length - 1" />
		</span>
	</div>
</template>
