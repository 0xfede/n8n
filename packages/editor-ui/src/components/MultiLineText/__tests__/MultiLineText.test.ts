import { render } from '@testing-library/vue';

import MultiLineText from '../MultiLineText.vue';

describe('MultiLineText', () => {
	it('renders multiline text correctly', () => {
		const { container } = render(MultiLineText, {
			props: {
				text: 'The quick brown fox\njumps over the lazy dog',
			},
		});

		expect(container).toMatchSnapshot();
	});
});
