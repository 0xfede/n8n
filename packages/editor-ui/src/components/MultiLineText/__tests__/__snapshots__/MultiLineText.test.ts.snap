// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultiLineText > renders multiline text correctly 1`] = `
<div>
  <div>
    
    <span>
      The quick brown fox
      <br />
    </span>
    <span>
      jumps over the lazy dog
      <!--v-if-->
    </span>
    
  </div>
</div>
`;
